app/lib/md.server.ts

```diff
@@ -49,20 +47,40 @@ export async function getProcessor(options?: ProcessorOptions) {
     import("rehype-slug"),
     import("rehype-stringify"),
     import("rehype-autolink-headings"),
+    import("rehype-pretty-code"),
     loadPlugins(),
   ]);
 
+  // The theme actually stores #FFFF${base-16-color-id} because vscode-textmate
+  // requires colors to be valid hex codes, if they aren't, it changes them to a
+  // default, so this is a mega hack to trick it.
+  const themeString = JSON.stringify(themeJson).replace(/#FFFF(.{2})/g, "var(--base$1)");
+  
   return unified()
     .use(remarkParse)
     .use(plugins.stripLinkExtPlugin, options)
-    .use(plugins.remarkCodeBlocksShiki, options)
     .use(remarkGfm)
     .use(remarkRehype, { allowDangerousHtml: true })
     .use(rehypeStringify, { allowDangerousHtml: true })
     .use(rehypeSlug)
-    .use(rehypeAutolinkHeadings);
+    .use(rehypeAutolinkHeadings)
+    .use(rehypePrettyCode, {
+      keepBackground: false,
+      theme: JSON.parse(themeString),
+      filterMetaString: (str) => str.replace(/lines=\[([^]*)\]/g, '{$1}').replace(/filename=([^ ]*)/g, 'title="$1"'),
+      transformers: [
+        transformerCopyButton({
+          visibility: 'always',
```

Comment:
Looks like this completely addresses my 2nd request

```suggestion
          visibility: 'hover',
```